bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/transceiver/EnOceanTransceiver.java

```diff
@@ -201,19 +235,26 @@ public void run() {
         logger.info("EnOceanSerialTransceiver RX thread started");
     }
 
-    public void ShutDown() {
+    public void shutDown() {
         logger.debug("shutting down transceiver");
         logger.debug("Interrupt rx Thread");
 
-        if (timeOut != null) {
-            timeOut.cancel(true);
+        Future<?> localTimeOut = timeOut;
+        if (localTimeOut != null) {
+            localTimeOut.cancel(true);
         }
 
-        if (readingTask != null) {
-            readingTask.cancel(true);
-            try {
-                inputStream.close();
-            } catch (Exception e) {
+        Future<?> localReadingTask = readingTask;
+        if (localReadingTask != null) {
+            localReadingTask.cancel(true);
+
+            InputStream localInputStream = inputStream;
+            if (localInputStream != null) {
+                try {
+                    localInputStream.close();
+                } catch (IOException e) {
+                    logger.debug("IOException occured while closing the stream", e);
```

Comment:
Yes, because this is such a large binding and this message could be duplicated elsewhere making it hard to troubleshoot. Putting the binding debug logging is usually per developer request. So i don't expect end-users logs to be bloated.